src/helpers/workspace.ts

```diff
@@ -185,3 +186,25 @@ async function selectTruffleConfigFromQuickPick(workspaces: TruffleWorkspace[]):
 
   return result.truffleWorkspace;
 }
+
+/**
+ * Every time the `workspace.onDidSaveTextDocument` listener emits a notification,
+ * this function receives, identifies the file extension and calls the corresponding function.
+ *
+ * @param globalState A memento object that stores state independent of the current opened workspace.
+ * @param document Represents a text document, such as a source file.
+ */
+export async function saveTextDocument(globalState: Memento, document: TextDocument): Promise<void> {
+  switch (path.extname(document.fileName)) {
+    case '.sol': {
```

Comment:
Might be cool to include Vyper (`.vy`) support here too, although it would definitely need wider consideration across the extension.